packages/node_modules/@webex/plugin-meetings/src/meeting/audio.js

```diff
@@ -0,0 +1,99 @@
+import StateMachine from 'javascript-state-machine';
+import StateMachineHistory from 'javascript-state-machine/lib/history';
+
+import Media from '../media';
+import MeetingUtil from '../meeting/util';
+import {MEETING_AUDIO_STATE_MACHINE, EVENT_TRIGGERS, EVENT_TYPES} from '../constants';
+
+const handleTransition = (value, audio, meeting) => {
+  if (audio.mute && audio.self) {
+    value = MEETING_AUDIO_STATE_MACHINE.STATES.MUTE_SELF;
+  }
+  if (!audio.mute && audio.self) {
+    value = MEETING_AUDIO_STATE_MACHINE.STATES.UNMUTE_SELF;
+  }
+  return value;
+};
+
+const AudioStateMachine = {
+  create() {
+    return new StateMachine({
+      transitions: [
+        {
+          name: MEETING_AUDIO_STATE_MACHINE.TRANSITIONS.INIT,
+          from: '*',
+          to: function (audio, meeting) {
+            return handleTransition(this.state, audio, meeting);
+          }
+        },
+        {
+          name: MEETING_AUDIO_STATE_MACHINE.TRANSITIONS.TOGGLE,
+          from: '*',
+          to: function (audio, meeting) {
+            return handleTransition(this.state, audio, meeting);
+          }
+        }
+      ],
+      data: {
+        muted: undefined,
+        self: undefined
+      },
+      methods: {
+        getMuted() {
+          return this.muted;
+        },
+        getSelf() {
+          return this.self;
```

Comment:
What is `self` here? I see it's set with `audio.self` but not sure what that is

Reply:
yeah may or may not need it, but it's to determine whether or not the action was remote or by self